includes/class-wc-adapter.php

```diff
@@ -0,0 +1,656 @@
+			ConfigurationInterface::MERCHANT_TOKEN         => @$this->gateway->merchant_token,
+			ConfigurationInterface::PAYEE_ID               => @$this->gateway->payee_id,
+			ConfigurationInterface::PAYEE_NAME             => get_bloginfo( 'name' ),
+			ConfigurationInterface::MODE                   => @$this->gateway->testmode === 'yes',
+			ConfigurationInterface::AUTO_CAPTURE           => @$this->gateway->auto_capture === 'yes',
+			ConfigurationInterface::SUBSITE                => @$this->gateway->subsite,
+			ConfigurationInterface::LANGUAGE               => @$this->gateway->culture,
+			ConfigurationInterface::SAVE_CC                => @$this->gateway->save_cc === 'yes',
+			ConfigurationInterface::TERMS_URL              => @$this->gateway->terms_url,
+			ConfigurationInterface::REJECT_CREDIT_CARDS    => @$this->gateway->reject_credit_cards === 'yes',
+			ConfigurationInterface::REJECT_DEBIT_CARDS     => @$this->gateway->reject_debit_cards === 'yes',
+			ConfigurationInterface::REJECT_CONSUMER_CARDS  => @$this->gateway->reject_consumer_cards === 'yes',
+			ConfigurationInterface::REJECT_CORPORATE_CARDS => @$this->gateway->reject_corporate_cards === 'yes',
+		];
+	}
+
+	/**
+	 * Get Platform Urls of Actions of Order (complete, cancel, callback urls).
+	 *
+	 * @param mixed $orderId
+	 *
+	 * @return array
+	 */
+	public function getPlatformUrls( $orderId ) {
+		$order = wc_get_order( $orderId );
+
+		if ( $this->gateway->is_new_credit_card ) {
+			return [
+				PlatformUrlsInterface::COMPLETE_URL => add_query_arg( 'action', 'swedbank_card_store',
+					admin_url( 'admin-ajax.php' ) ),
+				PlatformUrlsInterface::CANCEL_URL   => wc_get_account_endpoint_url( 'payment-methods' ),
+				PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+				PlatformUrlsInterface::TERMS_URL    => ''
+			];
+		}
+
+		if ( $this->gateway->is_change_credit_card ) {
+			return [
+				PlatformUrlsInterface::COMPLETE_URL => add_query_arg( [
+					'verify' => 'true',
+					'key'    => $order->get_order_key()
+				], $this->gateway->get_return_url( $order ) ),
+				PlatformUrlsInterface::CANCEL_URL   => $order->get_cancel_order_url_raw(),
+				PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+				PlatformUrlsInterface::TERMS_URL    => $this->getConfiguration()[ ConfigurationInterface::TERMS_URL ]
+			];
+		}
+
+		return [
+			PlatformUrlsInterface::COMPLETE_URL => $this->gateway->get_return_url( $order ),
+			PlatformUrlsInterface::CANCEL_URL   => $order->get_cancel_order_url_raw(),
+			PlatformUrlsInterface::CALLBACK_URL => WC()->api_request_url( get_class( $this->gateway ) ),
+			PlatformUrlsInterface::TERMS_URL    => $this->getConfiguration()[ ConfigurationInterface::TERMS_URL ]
+		];
+	}
+
+	/**
+	 * Get Order Data.
+	 *
+	 * @param mixed $orderId
+	 *
+	 * @return array
+	 */
+	public function getOrderData( $orderId ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method getOrderData() has an NPath complexity of 39285. The configured NPath complexity threshold is 200.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)